src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

```diff
@@ -106,6 +108,9 @@ public override void Abort(ConnectionAbortedException abortReason)
         // Only called after connection middleware is complete which means the ConnectionClosed token has fired.
         public override async ValueTask DisposeAsync()
         {
+            // Just in case we haven't started the connection, start it here so we can clean up properly.
+            EnsureStarted();
```

Comment:
I guess we can remove the `_receivingTask != null` and `_sendingTask != null` checks now. Technically these could still be null if some background thread flipped `_connectionStarted` and hasn't set these yet, but that would indicate some misusage of the ConnectionContext during dispose.

Reply:
Maybe it wouldn't be a misusage if the read or write was canceled before disposing.